src/components/table/column-picker/column-picker.component.ts

```diff
@@ -199,12 +208,13 @@ export class ColumnPickerComponent implements OnChanges {
         // remove each item from the selected columns list
         this.selected = this.selected.filter(column => columns.indexOf(column) === -1);
 
-        // add columns to deselected if not already there
+        const deselectedColumns = columns.filter(column => {
+            return this.deselected.indexOf(column) === -1 && this.deselected.indexOf({ name: column['name'], group: undefined}) === -1;
+        });
```

Comment:
This still doesn't look right, but you can probably fix this after resolving the above issues.